build fails looks likely real:

```
======================================================================
FAIL: test_update (corehq.apps.api.tests.TestCommCareUserResource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/api/tests.py", line 423, in test_update
    self.assertEqual(response.status_code, 202, response.content)
AssertionError: 200 != 202 : {"default_phone_number": "+50253311399", "email": "tlast@example.org", "first_name": "test", "groups": ["bdc1004b2a4e8ba605289034959a19a1"], "id": "bdc1004b2a4e8ba605289034954d618c", "language": "pol", "last_name": "last", "phone_numbers": ["+50253311399", "50253314588"], "resource_uri": "/a/qwerty/api/v0.5/user/bdc1004b2a4e8ba605289034954d618c/", "user_data": {"chw_id": "13/43/DFA"}, "username": "test"}
----------------------------------------------------------------------
Ran 85 tests in 539.934s
```

--- a/corehq/apps/api/es.py
+++ b/corehq/apps/api/es.py
@@ -195,7 +195,7 @@ class ESView(View):
         More powerful ES querying using POST params.
         """
         try:
-            raw_post = request.raw_post_data
+            raw_post = request.body
             raw_query = simplejson.loads(raw_post)
         except Exception, ex:
             content_response = dict(message="Error parsing query request", exception=ex.message)

--- a/corehq/apps/api/resources/v0_5.py
+++ b/corehq/apps/api/resources/v0_5.py
@@ -250,7 +250,7 @@ class GroupResource(v0_4.GroupResource):
         (BSD licensed) and modified to pass the kwargs to `obj_create` and support only create method
         """
         request = convert_post_to_patch(request)
-        deserialized = self.deserialize(request, request.raw_post_data, format=request.META.get('CONTENT_TYPE', 'application/json'))
+        deserialized = self.deserialize(request, request.body, format=request.META.get('CONTENT_TYPE', 'application/json'))
 
         collection_name = self._meta.collection_name
         if collection_name not in deserialized:
@@ -281,7 +281,7 @@ class GroupResource(v0_4.GroupResource):
         Exactly copied from https://github.com/toastdriven/django-tastypie/blob/v0.9.14/tastypie/resources.py#L1314
         (BSD licensed) and modified to catch Exception and not returning traceback
         """
-        deserialized = self.deserialize(request, request.raw_post_data, format=request.META.get('CONTENT_TYPE', 'application/json'))
+        deserialized = self.deserialize(request, request.body, format=request.META.get('CONTENT_TYPE', 'application/json'))
         deserialized = self.alter_deserialized_detail_data(request, deserialized)
         bundle = self.build_bundle(data=dict_strip_unicode_keys(deserialized), request=request)
         try:

--- a/corehq/apps/api/tests.py
+++ b/corehq/apps/api/tests.py
@@ -420,7 +420,7 @@ class TestCommCareUserResource(APIResourceTest):
         response = self.client.put(self.single_endpoint(backend_id),
                                    simplejson.dumps(user_json),
                                    content_type='application/json')
-        self.assertEqual(response.status_code, 202, response.content)
+        self.assertEqual(response.status_code, 200, response.content)
         self.assertEqual(1, len(CommCareUser.by_domain(self.domain.name)))
         modified = CommCareUser.get(backend_id)
         self.assertEqual(modified.username, "test")
@@ -551,7 +551,7 @@ class TestWebUserResource(APIResourceTest):
         response = self.client.put(self.single_endpoint(backend_id),
                                    simplejson.dumps(user_json),
                                    content_type='application/json')
-        self.assertEqual(response.status_code, 202, response.content)
+        self.assertEqual(response.status_code, 200, response.content)
         modified = WebUser.get(backend_id)
         self.assertEqual(modified.username, "test")
         self.assertEqual(modified.first_name, "Joe")
@@ -1117,7 +1117,7 @@ class TestGroupResource(APIResourceTest):
         response = self.client.put(self.single_endpoint(backend_id),
                                    simplejson.dumps(group_json),
                                    content_type='application/json')
-        self.assertEqual(response.status_code, 202, response.content)
+        self.assertEqual(response.status_code, 200, response.content)
         self.assertEqual(1, len(Group.by_domain(self.domain.name)))
         modified = Group.get(backend_id)
         self.assertEqual(modified.name, "test group")

--- a/corehq/apps/app_manager/views.py
+++ b/corehq/apps/app_manager/views.py
@@ -1862,7 +1862,7 @@ def edit_commcare_settings(request, domain, app_id):
 @require_can_edit_apps
 def edit_commcare_profile(request, domain, app_id):
     try:
-        settings = json.loads(request.raw_post_data)
+        settings = json.loads(request.body)
     except TypeError:
         return HttpResponseBadRequest(json.dumps({
             'reason': 'POST body must be of the form:'
@@ -1882,7 +1882,7 @@ def edit_commcare_profile(request, domain, app_id):
 
 
 def validate_langs(request, existing_langs, validate_build=True):
-    o = json.loads(request.raw_post_data)
+    o = json.loads(request.body)
     langs = o['langs']
     rename = o['rename']
     build = o['build']
@@ -1993,7 +1993,7 @@ def edit_app_attr(request, domain, app_id, attr):
     lang = request.COOKIES.get('lang', (app.langs or ['en'])[0])
 
     try:
-        hq_settings = json.loads(request.raw_post_data)['hq']
+        hq_settings = json.loads(request.body)['hq']
     except ValueError:
         hq_settings = request.POST
 

--- a/corehq/apps/cloudcare/views.py
+++ b/corehq/apps/cloudcare/views.py
@@ -456,7 +456,7 @@ class EditCloudcareUserPermissionsView(BaseUserSettingsView):
         }
 
     def put(self, request, *args, **kwargs):
-        j = json.loads(request.raw_post_data)
+        j = json.loads(request.body)
         old = ApplicationAccess.get_by_domain(self.domain)
         new = ApplicationAccess.wrap(j)
         old.restrict = new.restrict

--- a/corehq/apps/export/custom_export_helpers.py
+++ b/corehq/apps/export/custom_export_helpers.py
@@ -111,7 +111,7 @@ class CustomExportHelper(object):
     @property
     @memoized
     def post_data(self):
-        return json.loads(self.request.raw_post_data)
+        return json.loads(self.request.body)
 
     def update_custom_export(self):
         """

--- a/corehq/apps/reports/views.py
+++ b/corehq/apps/reports/views.py
@@ -486,7 +486,7 @@ def add_config(request, domain=None):
     from datetime import datetime
     user_id = request.couch_user._id
 
-    POST = json.loads(request.raw_post_data)
+    POST = json.loads(request.body)
     if 'name' not in POST or not POST['name']:
         return HttpResponseBadRequest()
 

--- a/corehq/apps/tropo/views.py
+++ b/corehq/apps/tropo/views.py
@@ -16,7 +16,7 @@ def sms_in(request):
     Handles tropo messaging requests
     """
     if request.method == "POST":
-        data = json.loads(request.raw_post_data)
+        data = json.loads(request.body)
         session = data["session"]
         # Handle when Tropo posts to us to send an SMS
         if "parameters" in session:
@@ -51,7 +51,7 @@ def ivr_in(request):
     Handles tropo call requests
     """
     if request.method == "POST":
-        data = json.loads(request.raw_post_data)
+        data = json.loads(request.body)
         phone_number = data["session"]["from"]["id"]
         # TODO: Implement tropo as an ivr backend. In the meantime, just log the call.
 

--- a/corehq/apps/users/views/__init__.py
+++ b/corehq/apps/users/views/__init__.py
@@ -632,7 +632,7 @@ def undo_remove_web_user(request, domain, record_id):
 def post_user_role(request, domain):
     if not domain_has_privilege(domain, privileges.ROLE_BASED_ACCESS):
         return json_response({})
-    role_data = json.loads(request.raw_post_data)
+    role_data = json.loads(request.body)
     role_data = dict([(p, role_data[p]) for p in set(UserRole.properties().keys() + ['_id', '_rev']) if p in role_data])
     role = UserRole.wrap(role_data)
     role.domain = domain

--- a/corehq/ex-submodules/couchforms/getters.py
+++ b/corehq/ex-submodules/couchforms/getters.py
@@ -43,7 +43,7 @@ def get_instance_and_attachment(request):
     else:
         # j2me and touchforms; of the form
         # $ curl --data '@form.xml' $URL
-        instance = request.raw_post_data
+        instance = request.body
         if not instance:
             instance = EMPTY_PAYLOAD_ERROR
     request._instance_and_attachment = (instance, attachments)

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -16,7 +16,7 @@ defusedxml==0.4.1
 diff-match-patch==20120106
 django-kombu==0.9.1
 django-rosetta==0.7.1
-django-tastypie==0.9.14
+django-tastypie==0.12.1
 djtables==0.1.1
 #egenix-mx-base==3.2.8
 eulxml==0.18.0

